source/javascripts/controllers/_EnvVarsController.js.erb

```diff
@@ -186,6 +199,45 @@ angular.module("BitriseWorkflowEditor").controller("EnvVarsController", function
 		return errors.length > 0 ? stringService.errorMessageFromErrors(errors) : null;
 	};
 
+	viewModel.valueVisibilityGetterSetterForEnvVar = function(secret) {
+		return function(shouldShowValue) {
+			if (shouldShowValue !== undefined) {
+				if (shouldShowValue && !secret.customProperties.progress.isInProgress) {
+					showValueOfSecret(secret);
+				}
+
+				secret.shouldShowValue = shouldShowValue;
+			}
+
+			return secret.shouldShowValue;
+		};
+	};
+
+	function showValueOfSecret(secret) {
+		secret.customProperties.progress.start("<%= data[:strings][:env_vars][:get_secret_env_value_progress][:in_progress] %>");
+
+		$q(function(resolve, reject) {
```

Comment:
The whole `$q(function` can now be replaced simply with the call itself:

`appService.getSecretValue(secret).then(`... set progress to succes/error